common/app/templates/inlineJS/blocking/shouldEnhance.scala.js

```diff
@@ -1,54 +1,98 @@
+            // if we're trying to set an old pref, set the new one
+            if ((settingCorePref = new RegExp(`^#${coreKey.replace('.', '\.')}=(on|off)$`).exec(hash)) && (corePref = settingCorePref[1])) {
+                hash = location.hash = `#${enhancedKey}=${corePref === 'off'}`;
+            }
+            // swap out the old temporary opt-in/out methods for the finalised ones
+            else if (/^#(featuresoff|core)$/.test(hash)) {
+                hash = location.hash = '#standard';
+            } else if (/^#(featureson|nocore)$/.test(hash)) {
+                hash = location.hash = '#enhanced';
+            }
         }
     };
 
-    // Guess whether the device is too old, regardless of whether it cuts the mustard
-    //
-    // 'older' iOS normally indicates a device with lower power (they stop being upgradeable at some point).
-    // We won't run all javascript on these.
-    // For usage stats see http://david-smith.org/iosversionstats/
-    //
-    // NOTE: this moves people into a category where they do not get important things such as commenting
-    var isOlderIOSDevice = function () {
-
-        if (navigator.platform === 'iPhone' || navigator.platform === 'iPad' || navigator.platform === 'iPod') {
-            // I'm intentionally being a bit over zealous in the detection department here
-            return /.*(iPhone|iPad; CPU) OS ([3456])_\d+.*/.test(navigator.userAgent);
-        }
+    function mustEnhance() {
+        if (hash === '#enhanced' || hash === `#${enhancedKey}=true`) return true;
+        if (preferEnhanced) return true;
         return false;
     };
 
-    var isIpad = function() {
-        return (navigator.platform === 'iPad');
+    function mustNotEnhance() {
+        return hash === '#standard' || hash === `#${enhancedKey}=false`;
     };
 
-    window.shouldEnhance = personDemandsFeatures() || (!personPrefersCore() && !isOlderIOSDevice() && !(@item.isFront && isIpad()));
-    window.shouldEnhance || console && console.info && console.info("THIS IS CORE");
-})(navigator, window);
+    function couldEnhance() {
+        return preferEnhanced !== false;
+    };
+
+    function weWantToEnhance() {
+        if (isOlderIOSDevice()) return false;
+        if (isFront) return !isIpad();
+        return true; // assume we want to enhance
+    };
+
+    // If this is an older iOS, we assume it's an older device (they stop being upgradeable at some point).
+    // Old here means iOS 3-6.
+    // For usage stats see http://david-smith.org/iosversionstats/
+    function isOlderIOSDevice() {
+        return /.*(iPhone|iPad; CPU) OS ([3456])_\d+.*/.test(navigator.userAgent);
+    };
+
+    function isIpad() {
+        return (platform === 'iPad');
+    };
+
+    // clean up old names before checking the ones we'll expect from now on
+    normaliseNomenclature();
+
+    // down to business
+    window.shouldEnhance = mustNotEnhance() ? false : mustEnhance() ? true : couldEnhance() && weWantToEnhance();
```

Comment:
I find this condition hard to get my head around. I'm not sure if this is correct/any better?

``` js
window.shouldEnhance = mustEnhance() || (couldEnhance() && weWantToEnhance() && !mustNotEnhance())
```